envoy/common/exception.h

```diff
@@ -12,6 +12,12 @@ class EnvoyException : public std::runtime_error {
   EnvoyException(const std::string& message) : std::runtime_error(message) {}
 };
 
+#define THROW_IF_NOT_OK(status_fn)                                                                 \
+  absl::Status status = status_fn;                                                                 \
+  if (!status.ok()) {                                                                              \
+    throw EnvoyException(std::string(status.message()));                                           \
+  }
```

Comment:
Consider adding this inside an inner scope, to allow reuse of `status` variable name in the same block.